File: 이지수/Python/Baekjoon_1034.py
# 1034
import sys
input = sys.stdin.readline

def getNum(line) :
	return int("0b" + "".join(line), 2)

def orders(lines) :
	order = {}	
	for line in lines :
		num = getNum(line) 
		if num in order : order[num] = (order[num][0]+1, line)
		else : order[num] = (1, line)
	return order

def isAvailable(line, isEnum) :
	count = line.count("0")
	return count <= k and count % 2 == isEnum 

def maxLineCount(orders) :
	for _, order in sorted(orders.items(), key=lambda item: item[1][0], reverse=True):
		if isAvailable(order[1], k % 2) : return order[0]
	return 0

n, m = map(int, input().split())
lines = [list(input().rstrip()) for _ in range(n)]
k = int(input())
print(maxLineCount(orders(lines)))

File: 이지수/Swift/Baekjoon_1034.swift
// 1034
import Foundation

@inline(__always)
func startWith(n: Int, m: Int) -> Void {
    var lines: [[Character]] = []
    for _ in 0..<n { lines.append(Array(readLine()!)) }
    let k: Int = Int(readLine()!)!
    print(solutionWith(orders: orders(by: lines), k: k))
}

@inline(__always)
func solutionWith(orders: [(Int, [Character])], k: Int) -> Int {
    for order in orders where isAvailable(line: order.1, k: k) { return order.0 }
    return 0
}

@inline(__always)
func num(line: [Character]) -> Int {
    return Int(line.reduce("") {"\($0)\($1)"}, radix:2)!
}

@inline(__always)
func orders(by lines: [[Character]]) -> [(Int, [Character])] {
    var orders: [Int: (Int, [Character])] = [:]
    for line in lines {
        let num: Int = num(line: line)
        orders[num] = ((orders[num]?.0 ?? 0) + 1, line)
    }
    return orders.values.sorted(by: {$0.0 > $1.0})
}

@inline(__always)
func isAvailable(line: [Character], k: Int) -> Bool {
    let count = line.filter{$0=="0"}.count
    return count <= k && count % 2 == k % 2
}

let input: [Int] = readLine()!.components(separatedBy: " ").map(){ Int($0)! }
startWith(n: input[0], m: input[1])
